mlrun/serving/states.py

```diff
@@ -1678,6 +1776,34 @@ class RootFlowStep(FlowStep):
     kind = "root"
     _dict_fields = ["steps", "engine", "final_step", "on_error"]
 
+    def __init__(
+        self,
+        name=None,
+        steps=None,
+        after: Optional[list] = None,
+        engine=None,
+        final_step=None,
+    ):
+        super().__init__(
+            name,
+            steps,
+            after,
+            engine,
+            final_step,
+        )
+        self._models = []
+
+    @property
+    def model_endpoints(self) -> list[str]:
+        return self._models
+
+    @model_endpoints.setter
+    def model_endpoints(self, models: list[str]):
+        self._models = models
+
+    def update_model_endpoint(self, model_endpoint_name: str):
+        self._models.append(model_endpoint_name)
```

Comment:
Relevant to the prev comment
```suggestion
    def update_model_endpoint(self, model_endpoint_names: list):
        self._models.extend(model_endpoint_names)
```

Reply:
Why not call this `add_model_endpoints`? After all, this doesn't really update any existing MEPs, it just adds more.